main.tf

```diff
@@ -220,6 +221,21 @@ resource "azurerm_kubernetes_cluster" "main" {
       condition     = !(var.microsoft_defender_enabled && !var.log_analytics_workspace_enabled)
       error_message = "Enabling Microsoft Defender requires that `log_analytics_workspace_enabled` be set to true."
     }
+    precondition {
+      // either we have automatic upgrades disabled
+      condition = (var.automatic_channel_upgrade == null || (
+        // or we have set it to patch and only specify the minor version
+        contains(["patch"], var.automatic_channel_upgrade) &&
+        matches("^[0-9]{1,}\\.[0-9]{1,}$", var.kubernetes_version)
```

Comment:
Is there any special reason we should use this regex here instead of determining whether `var.kubernetes_version` is `null` or empty?

Reply:
Exactly. Simply checking if it's null is only viable if someone set `automatic_channel_upgrade=stable|rapid` since then AKS updates to new minor version too and noone only specifies the major version I guess. But with automatic patch upgrades we want to make sure that a valid minor version (e.g `1.24` or `1.25`) is specified.

Reply:
Since I'm not an aks expert, I'll leave the shoot to @zioproto , you make the call.